packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +189,176 @@ class Tooltip extends mixin(
     this.tooltip.changeState(
       state,
       Object.assign(detail, { delegatorNode: this.element }),
-      callback
+      () => {
+        // @questions
+        // -- What is the expected behavior if the user clicks on the trigger when the tooltip is already open
+        // -- What's the focus state look like when there is only rich text within a tooltip (no buttons/links/etc)
+        // -- There is a "flash" where the next element (another trigger in the demo) is focused before focus is
+        //      moved to the tooltip. I'm assuming this is due to the debounce. Now that it's triggered on "click"
+        //      for keyboard users do we still want the debounce?
+        const isHidden = state !== 'shown';
+        this.tooltip.element.setAttribute('aria-hidden', isHidden.toString());
+        this.element.setAttribute('aria-expanded', (!isHidden).toString());
+
+        if (this._handleFocusinListener) {
+          this._handleFocusinListener = this.unmanage(
+            this._handleFocusinListener
+          ).release();
+        }
+
+        if (this._handleClickListener) {
+          this._handleClickListener = this.unmanage(
+            this._handleClickListener
+          ).release();
+        }
+
+        if (this._handleKeydownListener) {
+          this._handleKeydownListener = this.unmanage(
+            this._handleKeydownListener
+          ).release();
+        }
+
+        if (state === 'shown') {
```

Comment:
If this `if()` block is solely for focus management, existing code does it in the following manner:

1. When tooltip body is shown: At https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/components/src/components/floating-menu/floating-menu.js#L279-L285. Do you want to enhance it to support "primary focus or first tabbable" like https://github.com/carbon-design-system/carbon-custom-elements/blob/master/src/components/modal/modal.ts#L151-L163?
2. When tooltip body is hidden: At https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/components/src/components/floating-menu/floating-menu.js#L147-L154. Did you need any enhancement beyond the "focusing back to trigger" logic seen there?

Reply:
I moved the logic for primary focus and first tabbable to the floating menu as I do agree that this is not unique to tooltip and would benefit any floating menu.  I'm am doing more than just moving focus in the callback however because by default floating menu events are all attached to focus/blur events where with tooltip we need them to be solely on click. This caused me to to have to add in this additional hooks functions and then events on the the owner document.  Without them, I lost all of that capability.

Reply:
By moving this code to the floating menu, i'm not getting callstack errors when I open a second tooltip

Reply:
Thank you for the update! Wrt the remaining logic - Please see my other comments on "close-on-click-outside" logic.